File: molfinder/Models/MoleculeDataSource.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Szhchem.Data;

namespace molfinder.Models
{
    public class MoleculeDataSource
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        public long SubstanceID { get; set; }
        public long MolID { get; set; }
        public string ExtUrl { get; set; }
        public string ExtID { get; set; }
        public int DisplayOrder { get; set; }
        public int CompanyIntId { get; set; }
        public int DataSourceIntId { get; set; }
        public int UserRating { get; set; }
        public int Viewed { get; set; }
        public DateTime TimeCreated { get; set; }
        public List<CompanyInfo> CompanyInformation { get; set; } = new List<CompanyInfo>();
        public List<DataSource> DataSources { get; set; } = new List<DataSource>();
        [BsonIgnore] public int DataSourceType;
    }
}

File: molfinder/ClientApp/src/models/MoleculeDataSource.ts
import CompanyInfo from "./CompanyInfo";
import DataSource from "./DataSource";

export default class MoleculeDataSource {
    public id: string;
    public substanceID: number;
    public molID: number;
    public extUrl: string;
    public extID: string;
    public displayOrder: number;
    public companyIntId: number;
    public dataSourceIntId: number;
    public userRating: number;
    public viewed: number;
    public timeCreated: Date;
    public companyInformation: Array<CompanyInfo> = new Array<CompanyInfo>();
    public dataSources: Array<DataSource> = new Array<DataSource>();
    public dataSourceType: number;

    constructor(id: string, substanceID: number, molID: number, extUrl: string, extID: string, displayOrder: number, companyIntId: number, dataSourceIntId: number, userRating: number, viewed: number, timeCreated: Date, companyInformation: Array<CompanyInfo>, dataSources: Array<DataSource>, dataSourceType: number) {
        this.id = id;
        this.substanceID = substanceID;
        this.molID = molID;
        this.extUrl = extUrl;
        this.extID = extID;
        this.displayOrder = displayOrder;
        this.companyIntId = companyIntId;
        this.dataSourceIntId = dataSourceIntId;
        this.userRating = userRating;
        this.viewed = viewed;
        this.timeCreated = timeCreated;
        this.companyInformation = companyInformation;
        this.dataSources = dataSources;
        this.dataSourceType = dataSourceType;
    }
}